packages/preact/src/hydrated-component/index.tsx

```diff
@@ -39,14 +39,15 @@ onDocumentReady(() => {
       // If you forget to register a component with `registerHydratedComponent`, this is going to hang.
       const Component = await registry.getComponent(componentName);
 
-      const dataElement = el.querySelector('script[data-component-props]');
+      const dataElement = el.previousElementSibling;
```

Comment:
This relies on element ordering in the DOM. I don't anticipate problems with this, but let me know if this seems too clever.

Reply:
This seems fine, but I would appreciate a marker on the script tag that indicates what component the data is for.